pure-selenium/src/main/java/org/example/pages/InventoryPage.java

```diff
@@ -4,41 +4,87 @@
 import org.openqa.selenium.support.FindBy;
 import org.openqa.selenium.support.PageFactory;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
+import org.openqa.selenium.support.ui.Select;
 
-import java.util.List;
+public class InventoryPage extends AbstractPage{
 
-public class InventoryPage {
 
-    private WebDriver driver;
-    private static final Logger logger = LoggerFactory.getLogger(InventoryPage.class);
+    @FindBy(xpath = "//span[text()='Products']")
+    private WebElement pageTitle;
 
+    @FindBy(id = "react-burger-menu-btn")
+    private WebElement menuButton;
 
-    @FindBy(css = ".inventory_item_name")
-    private List<WebElement> productTitles;
+    @FindBy(id = "logout_sidebar_link")
+    private WebElement logoutLink;
 
-    @FindBy(css = "[data-test='product_sort_container']")
+    @FindBy(id = "add-to-cart-sauce-labs-backpack")
+    private WebElement addToCartButton;
+
+    @FindBy(className = "shopping_cart_badge")
+    private WebElement cartBadge;
+
+    @FindBy(id = "add-to-cart-sauce-labs-bike-light")
+    private WebElement addBikeLightToCartButton;
+
+    @FindBy(className = "shopping_cart_link")
+    private WebElement cartIcon;
+
+    @FindBy(className = "product_sort_container")
     private WebElement sortDropdown;
 
-    public InventoryPage(WebDriver driver) {
-        this.driver = driver;
-        PageFactory.initElements(driver, this);
+    @FindBy(className = "inventory_item_price")
+    private java.util.List<WebElement> productPrices;
+
+    public void sortByPriceHighToLow() {
+        Select select = new Select(sortDropdown);
+        select.selectByVisibleText("Price (high to low)");
+    }
+
+    public java.util.List<Double> getDisplayedPrices() {
+        return productPrices.stream()
+                .map(WebElement::getText)
+                .map(price -> price.replace("$", ""))
+                .map(Double::parseDouble)
+                .toList();
     }
 
-//    public List<WebElement> getProductTitles() {
-//        return productTitles;
-//    }
 
-    public void printAllProductTitles() {
-        if (productTitles.isEmpty()) {
-            logger.warn("No products found!");
+    public void clickCartIcon() {
+        cartIcon.click();
+    }
+
+    public void addBikeLightToCart() {
+        addBikeLightToCartButton.click();
+    }
+
+    public void addBackpackToCart() {
+        addToCartButton.click();
+    }
+
+    public int getCartCount() {
+        if (cartBadge.isDisplayed()) {
+            return Integer.parseInt(cartBadge.getText());
         } else {
-            logger.info("Products:");
-            for (WebElement title : productTitles) {
-                logger.info("- {}", title.getText());
-            }
+            return 0;
+        }
+    }
+
+    public void logout() {
+        menuButton.click();
+        try {
+            Thread.sleep(500);
+        } catch (InterruptedException e) {
+            e.printStackTrace();
```

Comment:
Using Thread.sleep is generally not recommended for waiting in web automation.  Instead, you can use explicit waits to ensure the elements are ready for interaction.